packages/widget/src/components/overlay-component/overlay-component.ts

```diff
@@ -0,0 +1,32 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { when } from 'lit/directives/when.js';
+import { styles } from './styles';
+
+@customElement('sygma-overlay-component')
+export class OverlayComponent extends LitElement {
+  static styles = styles;
+
+  @property({ type: Boolean })
+  isLoading = false;
+
+  render(): HTMLTemplateResult {
+    return when(
+      this.isLoading,
+      () => html`
+      <div class="loader">
+        <div class="overlay">
+          <div class="loading-spinner"></div>
```

Comment:
to be consistent with the naming it probably should be loadingSpinner

Reply:
but it'd be also good to switch to kebab case/BEM as soon as possible, agreed? definitely a separate PR material

Reply:
I have created this issue @sztok7